pkgs/development/python-modules/sphinxcontrib-ditaa/default.nix

```
@@ -0,0 +1,40 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, unittestCheckHook
+, mock
+, sphinx-testing
+, sphinx
+, ditaa
+}:
+
+buildPythonPackage rec {
+  pname = "sphinxcontrib-ditaa";
+  version = "1.0.2";
+  format = "setuptools";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "57f2e13b059b38fde89580beca916ac6ca7456b2d706c3ddea9dd4890e5f5bd3";
+  };
+
+  buildInputs = [ mock sphinx-testing ];
+  propagatedBuildInputs = [ sphinx ditaa ];
+
+  # no tests provided
+  doCheck = false;
+
+  nativeCheckInputs = [ unittestCheckHook ];
+
+  unittestFlagsArray = [ "-s" "tests" ];
```

```suggestion
  # no tests provided
  doCheck = false;
  
  pythonImportsCheck = [ "sphinxcontrib-ditaa" ];
```
Could you add `pythonImportsCheck` for a simple check?
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pythonimportscheck-using-pythonimportscheck

---

I've tried this -- it doesn't work out-of-the-box.   For sphinxcontrib-ditaa, get error like this:

```
AttributeError: module 'docutils.parsers.rst.directives' has no attribute 'images'
error: builder for '/nix/store/wkaisx7mz0l2a9yqlcisamp4g1rdfdaj-python3.11-sphinxcontrib-ditaa-1.0.2.drv' failed with exit code 1;
       last 10 log lines:
       >   File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
       >   File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
       >   File "<frozen importlib._bootstrap>", line 1147, in _find_and_load_unlocked
       >   File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
       >   File "<frozen importlib._bootstrap_external>", line 940, in exec_module
       >   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
       >   File "/nix/store/ja2lng4abns53sxa2x9hdhnrqpjr0k78-python3.11-sphinxcontrib-ditaa-1.0.2/lib/python3.11/site-packages/sphinxcontrib/ditaa.py", line 45, in <module>
       >     class DitaaDirective(directives.images.Image):
       >                          ^^^^^^^^^^^^^^^^^
       > AttributeError: module 'docutils.parsers.rst.directives' has no attribute 'images'
```

I can see images.py in nix store path for python3.11-docutils-0.21.1,  but adding it as dependency isn't enough to make it available to pythonImportsCheck.

I also reviewed all the other sphinxcontrib-xxx default.nix files.    Of those,   there are about a dozen that don't supply pythonImportsCheck:

xxx in blockdiag, httpdomain, plantuml:   pass,  will submit separate PRs to add pythonImportsCheck.
xxx in apidoc, applehelp, devhelp, htmlhelp, websupport, newsfeed, qthelp, jsmath, serializinghtml:  fail,  sometimes similar symptoms to sphinxcontrib-ditaa;  sometimes a problem with circular references.

---

Although sphinxcontrib-ditaa fails pythonImportsCheck,  it still works (as I assume the other sphinxcontrib modules do) when used from an environment alongside sphinx.  Maybe python module wrapping magic has something to do with this...

---

Thanks for investigating.
I'm not sure, but sphinx may add some handling.